[Feature] 사용자 지역 설정 추가 및 사용자 지역에 따른 날씨 정보 반환

walwaljj opened the pull request:
## 🔥 Related Issue

close: #22 

## 📝 Description

1. 지역을 저장합니다.
<img width="481" alt="image" src="https://github.com/OvercomingRoom/Bellbell/assets/108582847/af62d174-6bbe-4d8c-84cd-be13ffff93fe">

> 결과
> <img width="481" alt="image" src="https://github.com/OvercomingRoom/Bellbell/assets/108582847/3dfabfbb-b69c-4881-93e0-96a1080ac536">

2. 저장된 지역의 x, y 좌표에 따라 날씨를 반환합니다.
<img width="500" alt="image" src="https://github.com/OvercomingRoom/Bellbell/assets/108582847/d0d0e9b1-9b1e-4b72-8ff2-1ddbb5599027">

## ⭐️ Review

walwaljj committed:
:sparkles: feature : 사용자 지역 설정 추가 및 사용자 지역에 따른 날씨 정보 반환

IngyuSong commented on src/main/java/com/overcomingroom/bellbell/weather/service/WeatherService.java:
```diff
@@ -35,50 +37,88 @@ public class WeatherService {
 
         // 1. 사용자 정보 찾기
-        Member member = memberRepository.findByNickname(nickname).orElseThrow(() -> new CustomException(ErrorCode.MEMBER_INVALID));
+        Member member = findMemberByNickname(nickname);
```
nickname은 중복일 가능성이 있으므로 email 로 바꾸는 것이 좋을 것 같습니다.

walwaljj replied:
오 그러네용 수정하겠습니다!

walwaljj committed:
:recycle: refactor :
- 1. 사용자 정보 조회 파라미터 변경 (nickname -> email)
- 2. api url 변경 (단기 예보 -> 초 단기 예보)
- 3. api 변경에 따른 category 추가

walwaljj commented:
수정 완료했습니당 ㅎㅎ

IngyuSong reviewed (approved):
고생하셨습니다.
